### PR TITLE
Add `--prefer-index` flag for`imagetools create` on a single source

### PR DESCRIPTION
**Pull request by rvoh-tismith**
- Fixes issue #2481 
- I updated the tests that were affected by this change to test the new logic, however there were some tests in the repo that seem to have already been failing before I made any changes, so the few tests that are failing shouldn't have anything to do with my changes
- As a response to comments below, the default behavior isn't being changed -- the new logic is now behind the flag `--prefer-index=false`
- As a response to the comments below, I removed the mediaType check that I originally just added to because I can't really see a reason for it at this point
- Updated docs

### CODE REVIEW

File: util/imagetools/create.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func (r *Resolver) Combine(ctx context.Context, srcs []*Source, ann []string) ([
 
 	// on single source, return original bytes
 	if len(srcs) == 1 && len(ann) == 0 {
-		if mt := srcs[0].Desc.MediaType; mt == images.MediaTypeDockerSchema2ManifestList || mt == ocispec.MediaTypeImageIndex {
+		if mt := srcs[0].Desc.MediaType; mt == images.MediaTypeDockerSchema2ManifestList || mt == images.MediaTypeDockerSchema2Manifest || mt == ocispec.MediaTypeImageIndex {
```
Comment (tonistiigi) [inline util/imagetools/create.go:82]: If that is the desired behavior then why is there a mediatype check at all, or at least why is `ocispec.MediaTypeImageManifest` missing?

**Review by tonistiigi — COMMENTED** [review]
While I tend to agree that this makes more sense as a default behavior, this is removing the capability to convert image manifest into manifest list. Should we add a new flag that keeps previous behavior?

**Comment by rvoh-tismith** [comment]
> While I tend to agree that this makes more sense as a default behavior, this is removing the capability to convert image manifest into manifest list. Should we add a new flag that keeps previous behavior?

@tonistiigi great point. I was too focused on my own use case, but yeah, I definitely agree that some people are probably using this explicitly hoping to convert to a manifest list, or at the very least they've probably built workflows that are expecting this behavior, so changing the default behavior is probably too risky.

So, 2 things:

1. I think a flag for this behavior is the right way to go. I'm not married to the name, but maybe something like `--preserve-single-source`, so `docker buildx imagetools create --preserve-single-source=true -t foo/bar:newimage foo/bar:oldimage`.
2. Like I mentioned above, it feels like the behavior here should somewhat align with the behavior of `docker buildx build` which only outputs a manifest list if you specify multiple platforms. What are your thoughts there? Do you think there needs to be some alignment, or maybe the same flag for `build` so that you can control this behavior in both places? (that doesn't necessarily have to be included in this pr)

Thoughts?

File: util/imagetools/create.go (hunk 1 above)
Comment (rvoh-tismith) [reply util/imagetools/create.go:82]: Ahhhhh great point! Yeah, it actually seems reasonable that there doesn't even need to be a check. Thinking through the logic, I guess I can't really think of when the logic would need to be different here. Seems like if there is one source and no annotations (and possibly if the `--preserve-single-source` flag is set to `true`, based on my response to your other comment), we should just return the original bytes. I almost feel like I'm missing something here though. It feels like this is here for a reason, but maybe it is just an oversight.

**Comment by tonistiigi** [comment]
Maybe smth. like `--prefer-manifest-index` `--create-index` etc. for the behavior of always creating the index result and without the flag, it is the original manifest on a single source.

**Comment by rvoh-tismith** [comment]
> Maybe smth. like `--prefer-manifest-index` `--create-index` etc. for the behavior of always creating the index result and without the flag, it is the original manifest on a single source.

Nice. I'll make some changes and push them up.

**Comment by rvoh-tismith** [comment]
I split the difference and landed on `--prefer-index` lol. I also removed the mediaType checks.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**